### PR TITLE
Add parse abi method across deployers 

### PR DESCRIPTION
**Pull request by sczembor**
Closes #x

## Proposed Changes

- add new method `parseABIContractFile` across all deployers 
- create test scenario for it 

## TODO 
- [x] runtime
- [x] deployer
- [x] update change log

### CODE REVIEW

File: packages/algob/src/internal/deployer.ts [hunk 1]
```diff
@@ -581,6 +585,27 @@ class DeployerBasicMode {
 			return this.sendAndWait(Uint8ArraySignedTx, rounds);
 		}
 	}
+
+	/**
+     * Parses the file and return the ABIContract where the networks property
+	 *  is set to the currently used netowrk in deployer
+	 * @param fileName string
```
Comment (rmeena840) [inline packages/algob/src/internal/deployer.ts:592]:
```suggestion
	 * @param pathToFile string
```

File: packages/runtime/src/runtime.ts [hunk 2]
```diff
@@ -1186,6 +1189,20 @@ export class Runtime {
 		return this.executeTx(transactions);
 	}
 
+	/**
+	* Parses the file and return the ABIContract in case of network.runtime not defined throw exception
+	 * @param fileName string
```
Comment (rmeena840) [inline packages/runtime/src/runtime.ts:1194]:
```suggestion
	 * @param pathToFile string
```

File: packages/runtime/test/src/runtime.ts [hunk 3]
```diff
@@ -1441,13 +1442,56 @@ describe("Helper functions", function () {
...
+	const networkRuntimeNotDefined = "network-runtime-undefined.json";
+	const correctABI = "correct-ABI.json";
+	useFixture("atomic-transaction-composer")
 
 	this.beforeEach(function () {
 		runtime = new Runtime([]);
 	});
 
+	it("Should throw an error if file is not found", function () {
+		assert.throws(() => {
```
Comment (rmeena840) [inline packages/runtime/test/src/runtime.ts:1459]: use expectRuntimeError for checking the exact thrown error.

File: packages/runtime/test/src/runtime.ts [hunk 4]
```diff
@@ -1441,13 +1442,56 @@ describe("Helper functions", function () {
...
+		})
+	});
+  
+	it("Should throw an error if file is not json", function () {
+		assert.throws(() => {
+			runtime.parseABIContractFile(notJSON);
+		});
+	});
+  
+	it("Shoult from an error if network.runtime is not defined", function () {
```
Comment (rmeena840) [inline packages/runtime/test/src/runtime.ts:1476]: Would be great, if we can create runtime error for all these scenarios. It would be easy to catch them.

**Review by rmeena840 — COMMENTED** [review]
left comments.

File: packages/runtime/test/src/runtime.ts (hunk 3 above)
Comment (sczembor) [reply packages/runtime/test/src/runtime.ts:1459]: this error is not thrown by us. Its from external package

Comment (rmeena840) [reply packages/runtime/test/src/runtime.ts:1459]: ok

File: packages/runtime/test/fixtures/atomic-transaction-composer/not-JSON.teal [hunk 5]
```diff
@@ -0,0 +1,3 @@
+#pragma version 6
+int 1
+return
```
Comment (sczembor) [inline packages/runtime/test/fixtures/atomic-transaction-composer/not-JSON.teal:3]:
```suggestion
return

```

File: packages/runtime/test/fixtures/atomic-transaction-composer/not-ABI.json [hunk 6]
```diff
@@ -0,0 +1 @@
+{"name":"John", "age":30, "car":null}
```
Comment (sczembor) [inline packages/runtime/test/fixtures/atomic-transaction-composer/not-ABI.json:1]:
```suggestion
{"name":"John", "age":30, "car":null}

```

File: packages/runtime/test/fixtures/atomic-transaction-composer/network-runtime-undefined.json [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    "methods":[
+        {
+            "name":"add",
+            "desc":"Add 2 integers",
+            "args":[ { "type":"uint64" }, { "type":"uint64" } ],
+            "returns": {"type":"uint64"}
+        }
+    ]
+}
```
Comment (sczembor) [inline packages/runtime/test/fixtures/atomic-transaction-composer/network-runtime-undefined.json:16]:
```suggestion
}

```

File: packages/runtime/test/fixtures/atomic-transaction-composer/correct-ABI.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    "methods":[
+        {
+            "name":"add",
+            "desc":"Add 2 integers",
+            "args":[ { "type":"uint64" }, { "type":"uint64" } ],
+            "returns": {"type":"uint64"}
+        }
+    ]
+}
```
Comment (sczembor) [inline packages/runtime/test/fixtures/atomic-transaction-composer/correct-ABI.json:16]:
```suggestion
}

```

File: packages/algob/test/fixture-projects/abi-contract/basic-abi.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+    },
+    "methods":[
+        {
+            "name":"add",
+            "desc":"Add 2 integers",
+            "args":[ { "type":"uint64" }, { "type":"uint64" } ],
+            "returns": {"type":"uint64"}
+        }
+    ]
+}
```
Comment (sczembor) [inline packages/algob/test/fixture-projects/abi-contract/basic-abi.json:16]:
```suggestion
}

```

File: packages/web/src/lib/constants.ts [hunk 10]
```diff
@@ -7,3 +7,11 @@ export const ALGORAND_SIGN_TRANSACTION_REQUEST = "algo_signTxn";
...
 export const testnetURL = "https://node.testnet.algoexplorerapi.io";
 export const betanetURL = "https://node.betanet.algoexplorerapi.io";
+
+//official networks https://developer.algorand.org/docs/get-details/algorand-networks/
+export const mainnetGenesisHash = "wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=";
+export const testnetGenesisHash = "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=";
+export const betanetGenesisHash = "mFgazF+2uRS1tMiL9dsj01hJGySEmPN28B/TjjvpVW0=";
+
+//runtime network
+export const runtimeGenesisHash = "runtime";
```
Comment (robert-zaremba) [inline packages/web/src/lib/constants.ts:17]: why not defining it as upper case?

File: packages/web/src/lib/utils.ts [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+	testnetGenesisHash,
+} from "./constants";
+
+/**
+ * Queris the genesish hash based on the network name
+ * @param networkName string name of the network
+ * @returns if the network is known the genesisHash otherwise returns undefined
+ */
+export function getGenesisHashFromName(networkName: string): string | undefined {
+	switch (networkName) {
```
Comment (robert-zaremba) [inline packages/web/src/lib/utils.ts:14]:
```suggestion
	switch (networkName.toLowerCase()) {
```

File: packages/web/src/lib/constants.ts [hunk 12]
```diff
@@ -7,3 +7,11 @@ export const ALGORAND_SIGN_TRANSACTION_REQUEST = "algo_signTxn";
 export const mainnetURL = "https://node.algoexplorerapi.io";
 export const testnetURL = "https://node.testnet.algoexplorerapi.io";
 export const betanetURL = "https://node.betanet.algoexplorerapi.io";
+
+//official networks https://developer.algorand.org/docs/get-details/algorand-networks/
+export const mainnetGenesisHash = "wGHE2Pwdvd7S12BL5FaOP20EGYesN73ktiC1qzkkit8=";
+export const testnetGenesisHash = "SGO1GKSzyE7IEPItTxCByw9x8FmnrCDexi9/cOUJOiI=";
+export const betanetGenesisHash = "mFgazF+2uRS1tMiL9dsj01hJGySEmPN28B/TjjvpVW0=";
+
```
Comment (robert-zaremba) [inline packages/web/src/lib/constants.ts:15]: Add changelog:
+ list new functions 
+ list new constants
+ describe new functionality

**Review by robert-zaremba — APPROVED** [review]
lgtm

File: packages/runtime/src/interpreter/interpreter.ts [hunk 13]
```diff
@@ -55,7 +55,7 @@ export class Interpreter {
 	length: number; // total length of 'compiled' TEAL code
 	// local stores for a transaction.
 	bytecblock: Uint8Array[];
-	intcblock: BigInt[];
+	intcblock: bigint[];
```
[the remark this answers is not in the document]
Comment (sczembor) [reply packages/runtime/src/interpreter/interpreter.ts:58]: exactly

File: packages/web/src/lib/constants.ts (hunk 10 above)
Comment (sczembor) [reply packages/web/src/lib/constants.ts:17]: you mean the name of the const? `RUNTIME_GENESIS_HASH`?

Comment (robert-zaremba) [reply packages/web/src/lib/constants.ts:17]: yes, but maybe the lower case is more readable.

**Pull request merged**